aquarius/events/metadata_updater.py

```diff
@@ -733,22 +753,32 @@ def process_pool_events(self):
             logger.warning(f"exception thrown reading last_block from db: {e}")
             last_block = 0
 
-        block = self._web3.eth.blockNumber
-        if not block or not isinstance(block, int) or block <= last_block:
+        current_block = self._web3.eth.blockNumber
+        if (
+            not current_block
+            or not isinstance(current_block, int)
+            or current_block <= last_block
+        ):
             return
 
         from_block = last_block
-        logger.debug(
-            f"Price/Liquidity monitor >>>> from_block:{from_block}, current_block:{block} <<<<"
-        )
 
         start_block_chunk = from_block
-        for end_block_chunk in range(from_block, last_block + 1000, 1000):
+        for end_block_chunk in range(
+            from_block, current_block, self._blockchain_chunk_size
```

Comment:
i believe if you range to current_block + self._blockchain_chunk_size, you don't need to do the "last few blocks" processing (line 773)

Reply:
but if this was the bug, sorry about that 😟